src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.15;
+
+import "./ZkBobPool.sol";
+
+/**
+ * @title ZkBobCompoundingMixin
+ */
+abstract contract ZkBobCompoundingMixin is ZkBobPool {
+    using SafeERC20 for IERC20;
+
+    uint256 public investedAssetsAmount;
+
+    struct YieldParams {
+        // ERC4626 vault address (or address(0) if not set)
+        address yield;
+        // maximum amount of underlying tokens that can be invested into vault
+        uint256 maxInvestedAmount;
+        // expected amount of underlying tokens to be left at the pool after successful rebalance
+        uint256 buffer;
+        // slippage/rounding protection buffer, small part of accumulated interest that is non-claimable
+        uint96 dust;
+        // address to receive accumulated interest during the rebalance
+        address interestReceiver;
+        // operator address (or address(0) if permissionless)
+        address yieldOperator;
+    }
+
+    YieldParams public yieldParams;
+
+    event UpdateYieldParams(YieldParams yieldParams);
+    event Claimed(address indexed token, address indexed yield, uint256 amount);
+    event Rebalance(uint256 toPool, uint256 toYield);
+
+    // @inheritdoc ZkBobPool
+    function _withdrawToken(address _user, uint256 _tokenAmount) internal override {
+        uint256 underlyingBalance = IERC20(token).balanceOf(address(this));
+        if (underlyingBalance < _tokenAmount) {
+            YieldParams storage params = yieldParams;
+            IERC4626 yieldVault = IERC4626(params.yield);
+            uint256 buffer = params.buffer;
+            uint256 remainder = _tokenAmount - underlyingBalance;
+            uint256 vaultAmount = investedAssetsAmount;
+            if (vaultAmount >= remainder + buffer) {
+                investedAssetsAmount = vaultAmount - remainder - buffer;
+                yieldVault.withdraw(remainder + buffer, address(this), address(this));
+            } else {
+                investedAssetsAmount = vaultAmount - remainder;
+                yieldVault.withdraw(remainder, address(this), address(this));
+            }
+        }
+        IERC20(token).safeTransfer(_user, _tokenAmount);
+    }
+
+    /**
+     * @dev Updates yield parameters.
+     * Callable only by the contract owner / proxy admin.
+     * @param _yieldParams new yield parameters.
+     */
+    function updateYieldParams(YieldParams memory _yieldParams) external onlyOwner {
```

Comment:
What should we do with remaining tokens (claimable interest) in the vault, in case `_yieldParams.yield == address(0) && investedAssetsAmount == 0`?